i6_models/parts/conformer/mhsa.py

```diff
@@ -0,0 +1,42 @@
+from __future__ import annotations
+from dataclasses import dataclass
+from typing import Optional, Callable
+import torch
+
+from i6_models.config import ModelConfiguration
+
+
+@dataclass
+class ConformerMHSAV1Config(ModelConfiguration):
+    input_dim: int  # input dim and total dimension for query/key and value projections, should be dividable by `num_att_heads`
+    num_att_heads: int  # number of attention heads
+    key_padding_mask: Optional[
```

Comment:
I think this has to be input to the `forward` and not part of the Config since this depends on padding/batching, so it must not be static. In [torchaudio](https://pytorch.org/audio/main/generated/torchaudio.models.Conformer.html) they call it `lengths`.